src/kw/kw.cpp

```diff
@@ -156,35 +224,164 @@ string KanoWorld::get_refresh_header(string token)
 
 
 /**
- *   libcurl callback function to collect response from the server
+ * \brief Returns wether the user is currently logged in
  *
+ * A local token timestamp comparison is performed, to decide wether it has
+ * expired
+ *
+ * \param verbose    Be more descriptive on the actions performed
+ *
+ * \returns True if the current user is logged in.
  */
-size_t KanoWorld::write_function(void *ptr, size_t size, size_t nmemb, void *stream)
+bool KanoWorld::is_logged_in(const bool verbose)
```

Comment:
On my branch, I added a utility function called `getTimestamp` which returns the number of milliseconds since last epoch which might be easier to use here. Also, `<chrono>` is the C++ way of working with time; might be an interesting read.

Reply:
Sounds really good. Right now we'll stick with what we have but let's definitely migrate to use your stuff for the next release.